cpp/server/ct-server.cc

```diff
@@ -317,17 +317,23 @@ int main(int argc, char* argv[]) {
   // Publish this node's hostname:port info
   cluster_controller.SetNodeHostPort(FLAGS_server, FLAGS_port);
 
-  // Periodically refresh our contiguous tree size so that other nodes get a
-  // good view of the data we have locally and can use us for replication.
-  // This has the side effect of refreshing the TTL on our ClusterNodeState
-  // entry.
+  // Periodically refresh our ClusterNodeState entry so it doesn't get deleted.
+  // TODO(alcutter): Figure out if we actually need this here now, the signer
+  // thread above will refresh the entry, but potentially not frequently
+  // enough.
   PeriodicClosure node_state_refresh(
       event_base, std::chrono::seconds(FLAGS_node_state_refresh_seconds),
       [&db, &cluster_controller, &internal_pool]() {
         // Actually run this on the thread pool, since this periodic closure is
         // calling back on the eventloop thread.
         internal_pool.Add([&db, &cluster_controller]() {
-          cluster_controller.ContiguousTreeSizeUpdated(db->TreeSize());
+          SignedTreeHead latest_sth;
+          if (db->LatestTreeHead(&latest_sth) !=
+              Database<LoggedCertificate>::LOOKUP_OK) {
```

Comment:
Isn't this going to compete with the tree signer? If I'm not mistaken, the STH in the database is the serving STH, and the tree signer puts up the locally signed STH, no?